tests/python/gaia-ui-tests/gaiatest/apps/wallpaper/app.py

```diff
@@ -14,8 +14,9 @@ class Wallpaper(Base):
     _stock_wallpapers_locator = (By.CLASS_NAME, 'wallpaper')
 
     def tap_wallpaper_by_index(self, index):
-        self.wait_for_condition(lambda m: len(m.find_elements(*self._stock_wallpapers_locator)) >= index,
-            message = '%s Wallpapers not present after timeout' % index)
+        Wait(self.marionette).until(
+            lambda m: len(m.find_elements(*self._stock_wallpapers_locator)) >= (index + 1),
+            message='%d wallpaper(s) not present after timeout' % (index + 1))
```

Comment:
The original had ' >= index' and you're replacing it with >= (index + 1). Why is that?
It seems to work though with the test that's using tap_wallpaper_by_index (test_settings_wallpaper.py).